src/libs/pathfinding.cpp

```diff
@@ -1 +1,47 @@
-#include "pathfinding.h"
\ No newline at end of file
+#include "pathfinding.h"
+
+followlib::PathFinding::PathFinding(turtlesim::Pose goal, double d, ros::NodeHandle &n){
+  finalLocation = goal;
+  nh = n;
+  distance_tolerance = d;
+  cmd_vel_pub = nh.advertise<geometry_msgs::Twist>("/pathfinding/cmd_vel", 1);
+  pose_sub = n.subscribe("/pose_vel", 1000, &PathFinding::poseCallBack,this);
+
+}
+void followlib::PathFinding::poseCallBack(const turtlesim::Pose &p_msg){
+  initLocation.x = p_msg.x;
+  initLocation.y = p_msg.y;
+  initLocation.theta = p_msg.theta;
+}
+
+double followlib::PathFinding::getDistance(double x1, double y1, double x2,double y2){
+  return sqrt(pow((x1-x2),2) + pow((y1-y2),2));
+}
+
+void followlib::PathFinding::moveGoal(){
+  geometry_msgs::Twist vel_msg;
+
+  ros::Rate loop_rate(10);
+
+  do{
+    vel_msg.linear.x = 1.5*getDistance(initLocation.x,initLocation.y,finalLocation.x,finalLocation.y);
```

Comment:
Maybe it would be better to set this to a constant value, like .2 meters per second.